woodwork/type_sys/inference_functions.py

```diff
@@ -117,6 +117,7 @@ def __init__(self, get_regex: Callable[[], str]):
         self.get_regex = get_regex
 
     def __call__(self, series: pd.Series) -> bool:
+        series = series.dropna()
```

Comment:
Since we are adding this dropna, maybe we should add a test to make sure in our bad emails test that a series of all `nan` values does not get inferred as EmailAddress?